source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRLayoutHelper.mm

```diff
@@ -264,5 +286,239 @@ - (BOOL)shouldUseNewLayoutForView:(ACRView *)rootView
     return final_layout;
 }
 
+/**
+ * This methods distributes available width amongst all items of given container
+ */
+- (float)containerWidthDistributor:(std::shared_ptr<Container> const)container
+                    availableWidth:(float)availableWidth
+                     andHostConfig:(ACOHostConfig *)config
+{
+    std::shared_ptr<Layout> final_layout = [self layoutToApplyFrom:container->GetLayouts() andHostConfig:config];
+    if(final_layout->GetLayoutContainerType() == LayoutContainerType::Flow)
+    {
+        std::shared_ptr<FlowLayout> flow_layout = std::dynamic_pointer_cast<FlowLayout>(final_layout);
```

Comment:
we need to dynamically cast here, can we not use any better approach like deserialize